### PR TITLE
Update to support the new Group API and TileDB 2.8.0

### PR DESCRIPTION
**Pull request by DimitrisStaratzis**
Update to TileDB 2.8.0 and support for the new Group API. 

In this PR I commented out two tests regarding consolidation and vacuuming. These tests are failing or produce logically mistaken results because of the issues I highlight in these two stories:

[Vacuuming bug](https://app.shortcut.com/tiledb-inc/story/16943/vacuuming-fails-to-vacuum-fragments-in-given-timestamp-range)

[Consolidation bug](https://app.shortcut.com/tiledb-inc/story/16941/consolidation-fails-to-consolidate-fragments-in-given-timestamp-range )

In addition, the `group.deleteMetadata("key")` method is flaky producing wrong results sometimes. (FIXED) with 2.8.1
[Metadata bug](https://app.shortcut.com/tiledb-inc/story/16988/group-api-delete-metadata-key-method-flakiness)

### CODE REVIEW

File: src/main/java/io/tiledb/java/api/Group.java [hunk 1]
```diff
@@ -24,12 +24,403 @@
...
+  }
+
+  /**
+   * This method is responsible for closing the group object without deleting its pointers. It is
+   * useful for cases were we need to reopen a group. The "close" method above deletes a group
+   * completely.
+   *
+   * @throws TileDBError
+   */
+  public void sleep() throws TileDBError {
```
Comment (DimitrisStaratzis) [inline src/main/java/io/tiledb/java/api/Group.java:392]: This method is equivalent to the close() method in C++. However all objects in java need to implement the autoclosable interface which contains a "close()" method in which all pointers and the object itself is deleted. Close() in c++ has the meaning of sleep. This is the reasoning behind the naming here.

**Comment by ihnorton** [comment]
> method is flaky producing wrong results sometimes.

This could happen if the timestamps overlap. You may need to specify distinct timestamps to avoid overlap.

File: src/main/java/io/tiledb/java/api/Group.java [hunk 2]
```diff
@@ -24,12 +24,403 @@
...
 
-public class Group {
-
+public class Group implements AutoCloseable {
   private final Context ctx;
   private final String uri;
+  private SWIGTYPE_p_tiledb_group_t groupp;
+  private SWIGTYPE_p_p_tiledb_group_t grouppp;
+
+  public Group(Context ctx, String uri) throws TileDBError {
```
Comment (ihnorton) [inline src/main/java/io/tiledb/java/api/Group.java:46]: Let's make the constructor take a third parameter for the mode, like the C++ API.

File: src/main/java/io/tiledb/java/api/Group.java (hunk 1 above)
Comment (ihnorton) [reply src/main/java/io/tiledb/java/api/Group.java:392]: I don't think `sleep` is quite right here. `detach` or `release` or `disengage` seems closer to the meaning we need here.

Comment (ihnorton) [reply src/main/java/io/tiledb/java/api/Group.java:392]: Actually I think we should:
- hide `.open` and take the mode in the constructor as in the earlier comment (so `.open` would become private only)
- implement `reopen` instead, similar to the `Array` API, and then make this call to `tiledb_group_close` inside of `reopen`
eg:
```
void reopen(mode) throws TileDBError {
  tiledb_group_close(self)
  self._private_open(mode);
}
```

**Review by ihnorton — CHANGES_REQUESTED** [review]
Some changes related to open/close, because I think we should be more consistent with the Array API and avoid the potential confusion introducing a `sleep` function.

File: src/main/java/io/tiledb/java/api/Group.java (hunk 1 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/java/api/Group.java]: Done!

File: src/main/java/io/tiledb/java/api/Group.java (hunk 2 above)
Comment (DimitrisStaratzis) [reply src/main/java/io/tiledb/java/api/Group.java]: All done, thanks for the comments!

File: src/test/java/io/tiledb/java/api/GroupTest.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        (int[]) intArray.toJavaArray(), (int[]) metadataIntArray.toJavaArray());
+
+    // close group and reopen in write mode
+    config = new Config();
+    config.set("sm.group.timestamp_start", "20");
+    context = new Context(config);
+    group.reopen(context, TILEDB_WRITE);
+    // delete a metadata entry
+    group.deleteMetadata("key2");
+    // TODO this delete functionality does not always work. The core library does not have any tests
```
Comment (ihnorton) [inline src/test/java/io/tiledb/java/api/GroupTest.java:196]: Let's make this a conditional failure when libtiledb version is less than 2.8.0 -- should be fixed in 2.8.1

**Review by ihnorton — APPROVED** [review]
LGTM except for the one change to mark conditional failure, or you can wait to merge until 2.8.1 is released if you prefer.

**Comment by DimitrisStaratzis** [comment]
> > > method is flaky producing wrong results sometimes.
> > 
> > 
> > This could happen if the timestamps overlap. You may need to specify distinct timestamps to avoid overlap.
> 
> Not sure what you mean here. the Group API does not have any methods related to timestamps.

The issue was fixed with TileDB 2.8.1

**Pull request merged**